Mathlib/RingTheory/Extension/Cotangent/LocalizationAway.lean

```diff
@@ -139,4 +149,91 @@ lemma liftBaseChange_injective_of_isLocalizationAway :
     Module.End.one_apply, LinearMap.smul_apply, one_smul, Algebra.Extension.Cotangent.map_mk,
     Extension.Cotangent.mk_eq_zero_iff] using hx
 
+/--
+In the notation of the module docstring: Since `T` is standard smooth
+of relative dimension one over `S`, `K/K²` is free of rank one generated
+by the image of `g * X - 1`.
+This is the section `K/K² → J/J²` defined by sending the image of `g * X - 1` to `x : J/J²`.
+-/
+noncomputable def cotangentCompAwaySec (x : ((localizationAway T g).comp P).toExtension.Cotangent) :
+    (localizationAway T g).toExtension.Cotangent →ₗ[T]
+      ((localizationAway T g).comp P).toExtension.Cotangent :=
+  (basisCotangentAway T g).constr T fun _ ↦ x
+
+variable (x : ((localizationAway T g).comp P).toExtension.Cotangent)
+
+/-- By construction, the section `cotangentCompAwaySec` sends `g * X - 1` to `x`. -/
+lemma cotangentCompAwaySec_apply :
+    cotangentCompAwaySec g P x
+      (Extension.Cotangent.mk ⟨C g * X () - 1, C_mul_X_sub_one_mem_ker _⟩) = x := by
```

Comment:
I wonder if `Extension.Cotangent.mk ⟨C g * X () - 1, C_mul_X_sub_one_mem_ker _⟩` should be its own def.